examples/auto_deploy/README.md

```diff
@@ -261,19 +261,189 @@ For expert TensorRT-LLM users, we also expose the full set of [`LlmArgs`](../../
 - All config fields that are used by the AutoDeploy core pipeline (i.e. the `InferenceOptimizer`) are
   _exclusively_ exposed in the [`AutoDeployConfig` class](../../tensorrt_llm/_torch/auto_deploy/llm_args.py).
   Please make sure to refer to those first.
-- For expert users we expose the full set of [`LlmArgs`](../../tensorrt_llm/_torch/auto_deploy/llm_args.py#L201)
+- For expert users we expose the full set of [`LlmArgs`](../../tensorrt_llm/_torch/auto_deploy/llm_args.py#L222)
   that can be used to configure the [AutoDeploy `LLM` API](../../tensorrt_llm/_torch/auto_deploy/llm.py#L13) including runtime options.
-- Note that some fields in the full [`LlmArgs`](../../tensorrt_llm/_torch/auto_deploy/llm_args.py#L201)
+- Note that some fields in the full [`LlmArgs`](../../tensorrt_llm/_torch/auto_deploy/llm_args.py#L222)
   object are overlapping, duplicated, and/or _ignored_ in AutoDeploy, particularly arguments
   pertaining to configuring the model itself since AutoDeploy's model ingestion+optimize pipeline
   significantly differs from the default manual workflow in TensorRT-LLM.
-- However, with the proper care the full [`LlmArgs`](../../tensorrt_llm/_torch/auto_deploy/llm_args.py#L201)
+- However, with the proper care the full [`LlmArgs`](../../tensorrt_llm/_torch/auto_deploy/llm_args.py#L222)
   objects can be used to configure advanced runtime options in TensorRT-LLM.
 - Note that any valid field can be simply provided as keyword argument ("`**kwargs`") to the
   [AutoDeploy `LLM` API](../../tensorrt_llm/_torch/auto_deploy/llm.py#L13).
 
 </details>
 
+### Expert Configuration of `build_and_run_ad.py`
+
+For expert users, `build_and_run_ad.py` provides advanced configuration capabilities through a flexible argument parser powered by PyDantic Settings and OmegaConf. You can use dot notation for CLI arguments, provide multiple YAML configuration files, and leverage sophisticated configuration precedence rules to create complex deployment configurations.
+
+<details>
+<summary>Click to expand for detailed configuration examples</summary>
+
+#### CLI Arguments with Dot Notation
+
+The script supports flexible CLI argument parsing using dot notation to modify nested configurations dynamically. You can target any field in both the [`ExperimentConfig`](./build_and_run_ad.py#L92) and nested [`AutoDeployConfig`](../../tensorrt_llm/_torch/auto_deploy/llm_args.py#L41)/[`LlmArgs`](../../tensorrt_llm/_torch/auto_deploy/llm_args.py#L221) objects:
+
+```bash
+# Configure model parameters
```

Comment:
TODO for @lucaslie: mention that those are _not_ explicitly listed fields when running `python build_and_run_ad.py --help` since they are just nested dict values created on the fly